Packs/UrlScan/Integrations/UrlScan/UrlScan.py

```diff
@@ -570,32 +566,76 @@ def urlscan_submit_command(client):
-        if response.get('url_is_blacklisted') or response.get('is_error'):
-            execution_metrics.general_error += 1
-            if response.get('is_error'):
-                schedule_and_report(command_results=command_results, items_to_schedule=items_to_schedule,
-                                    execution_metrics=execution_metrics, rate_limit_reset_after=rate_limit_reset_after)
-                return_results(results=command_results)
-                return_error(response.get('error_string'))
-            continue
-        if metrics == ErrorTypes.QUOTA_ERROR:
-            if not is_scheduled_command_retry():
-                execution_metrics.quota_error += 1
-            items_to_schedule.append(url)
-            continue
-        uuid = response.get('uuid')
-        use_url_as_name = True if demisto.args()['use_url_as_name'] == 'true' else False
-        get_urlscan_submit_results_polling(client, uuid, use_url_as_name)
-        execution_metrics.success += 1
+    search_only = argToBoolean(demisto.args().get("search_only", False))
+    if search_only or IS_SYNC_MODE:
+        urlscan_search_only(client, command_results, execution_metrics)
+    else:
+        urls = argToList(demisto.args().get('url'))
+        for url in urls:
+            demisto.args()['url'] = url
+            response, metrics, rate_limit_reset_after = urlscan_submit_url(client, url)
+            if response.get('url_is_blacklisted') or response.get('is_error'):
+                execution_metrics.general_error += 1
+                if response.get('is_error'):
+                    schedule_and_report(command_results=command_results, items_to_schedule=items_to_schedule,
+                                        execution_metrics=execution_metrics, rate_limit_reset_after=rate_limit_reset_after)
+                    return_results(results=command_results)
+                    return_error(response.get('error_string'))
+                continue
+            if metrics == ErrorTypes.QUOTA_ERROR:
+                if not is_scheduled_command_retry():
+                    execution_metrics.quota_error += 1
+                items_to_schedule.append(url)
+                continue
+            uuid = response.get('uuid')
+            use_url_as_name = demisto.args()['use_url_as_name'] == 'true'
+            get_urlscan_submit_results_polling(client, uuid, use_url_as_name)
+            execution_metrics.success += 1
     schedule_and_report(command_results=command_results, items_to_schedule=items_to_schedule,
                         execution_metrics=execution_metrics, rate_limit_reset_after=rate_limit_reset_after)
     return command_results
 
 
+def urlscan_search_only(client: Client, command_results: list, execution_metrics: ExecutionMetrics):
+    urls = argToList(demisto.args().get("url"))
+    for url in urls:
+        demisto.args()["url"] = url
+        response = urlscan_search(client, "page.url", quote(url, safe=""))
```

Comment:
@anara123 since we're aiming to improve performance, it'll be best to introduce a ThreadPoolExecutor or such.

Reply:
@DeanArbel In sync-mode you don't really get a list of urls but only a single one, so it doesn't really matter

Reply:
@mmhw you are right. But there is a plan to pass a list of url in syncmode

Reply:
@DeanArbel @anara123 I added ThreadPoolExecutor